### PR TITLE
Add Devise

### PR DESCRIPTION
**Pull request by AbuRayhaan**
##**BLOG-APP: ADD DEVISE**

In this milestone, we implemented the following:

- `Installed and Setup Devise`.
- Removed `current_user method` in `ApplicationController`.
- Added `devise` to `route.rb`.
- Ensured that App can `register` new user with `email` and `password`.
- Allows users to ask for `confirmation mails`.
- Allows users to `reset password`.
- Documented the project in a `professional way`.
- Followed `Ruby best practices`.
- Used the `correct Github flow process`.
- Ensured there are `no linters error`.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (FrederickMih) [inline app/models/user.rb]: - [ ] Hi @AbuRayhaan, based on the requirements of this project, the application should request the user to confirm the password. To achieve this, `confirmable` should be enabled in the user model that's

```.ruby
 devise :database_authenticatable, :registerable,
         :recoverable, :rememberable, :validatable, :confirmable
```

File: app/views/partials/_header.html.erb [hunk 2]
```diff
@@ -1,3 +1,5 @@
 <header>
   <h1><%= link_to "BLOG APP", root_path%></h1>
+
+  <%= link_to "Sign Up"%>
```
Comment (FrederickMih) [inline app/views/partials/_header.html.erb]: - [ ] On your header, this sign-up link does not link to your sign-up page. Kindly make sure that a user can sign up and login into your app.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @AbuRayhaan,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights :green_circle: 

- PR has a title and summary :heavy_check_mark: 
- Correct GitHub workflow :heavy_check_mark: 
- No Linter error :heavy_check_mark: 
- Meaningful commit messages :heavy_check_mark: 
- Descriptive README.md file :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @AbuRayhaan , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Good points: 
- Device is properly setup 
- Users can register and log in

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**